frontend/utilities/yaml/index.ts

```diff
@@ -1,11 +1,38 @@
+import yaml from "js-yaml";
+
 interface IYAMLError {
   name: string;
   reason: string;
   line: string;
 }
 
-const constructErrorString = (yamlError: IYAMLError) => {
+export const constructErrorString = (yamlError: IYAMLError) => {
   return `${yamlError.name}: ${yamlError.reason} at line ${yamlError.line}`;
 };
 
+export const agentOptionsToYaml = (agentOpts: any) => {
```

Comment:
Currently this was done (or meant to be done, I guess) at the callsite with the `|| {}` but you're right, maybe it should be done here inside this function instead.

Reply:
Added the check inside the func. Regarding the type, `agent_options` is not strictly defined even on the backend, it can have arbitrary keys as those are passed to the osquery agent ultimately, so I left it as `any`.